wps-cloudformation-template.yaml

```diff
@@ -585,6 +585,84 @@ Resources:
                 Effect: "Allow"
                 Action: ["ses:SendRawEmail"]
                 Resource: ["*"]
+  EmptyBucketLambdaFunction:
+    Type: "AWS::Lambda::Function"
+    Condition: CreateEphemeralBucket
+    Properties:
+      Code:
+        ZipFile: |
+          import boto3
+          import cfnresponse
+
+          def handler(event, context):
+               try:
+                   if event['RequestType'] == 'Delete':
+                       s3 = boto3.resource('s3')
+                       bucket = s3.Bucket(os.getenv('BUCKET'))
+                       deleteBatch = []
+                       print "Deleting"
+                       for obj in bucket.objects.all():
```

Comment:
Sorry, after noting the pagination stuff, it seems the following is fine for non-versioned buckets, and handles all of the pagination for you:

```python
bucket.objects.all().delete()
```